addons/controllers/packageinstallstatus_controller.go

```diff
@@ -0,0 +1,442 @@
+		}
+		statusChanged = statusChanged || conditionChanged
+	}
+
+	if !statusChanged {
+		log.Info("no change in the reconciliation status of any of the core/additional packages in the cluster")
+		return nil
+	}
+
+	if err := r.Client.Status().Update(ctx, clusterBootstrap); err != nil {
+		return err
+	}
+
+	log.Info("Successfully reconciled ClusterBootstrapStatus")
+	return nil
+}
+
+// reconcileClusterBootstrapStatus reconciles clusterBootstrapStatus by setting conditions corresponding to all core/additional packages
+// The Status update happens in the caller function
+func (r *PackageInstallStatusReconciler) reconcileClusterBootstrapStatus(
+	ctx context.Context,
+	c client.Client,
+	clusterBootstrapStatus *runtanzuv1alpha3.ClusterBootstrapStatus,
+	clusterName string,
+	clusterRole ClusterRole,
+	pkgName string,
+	pkgiNamespace string,
+	log logr.Logger) (bool, error) {
+
+	var pkgiName, pkgShortname string
+
+	Client := c
+	// controller-runtime's client (r.Client) is not able to find aggregated api server resources.
+	// That's why we need to use r.aggregatedAPIResourcesClient for fetching packageMetadata inside the management cluster.
+	// This include packages belonging to the management cluster and workload clusters' kapp controller package which
+	// is installed in the workload cluster's namespace in the management cluster.
+	// For all other packages belonging to the workload cluster, remoteClient should be used for fetching packageMetadata
+	if clusterRole == clusterRoleManagement || strings.Contains(pkgName, kappCtrlPkgPrefix) {
+		Client = r.aggregatedAPIResourcesClient
+	}
+
+	// Note: pkgi name is determined from pkg.RefName. That's why we rely on the existence of package metadata here
+	pkgRefName, pkgVersion, err := util.GetPackageMetadata(ctx, Client, pkgName, pkgiNamespace)
+	if pkgRefName == "" || pkgVersion == "" || err != nil {
+		log.Error(err, fmt.Sprintf("unable to fetch Package.Spec.RefName or Package.Spec.Version from Package '%s/%s'", pkgiNamespace, pkgName))
+		return false, err
+	}
+	pkgShortname = strings.Split(pkgRefName, ".")[0]
+
+	// package install name for core/additional packages in the workload cluster follows the <cluster name>-<addon short name> naming convention
+	// for management cluster, the naming convention for core/additional package installs is just <addon short name>
+	if clusterRole == clusterRoleWorkload {
+		pkgiName = util.GeneratePackageInstallName(clusterName, pkgRefName)
+	} else if clusterRole == clusterRoleManagement {
+		pkgiName = pkgShortname
+	}
+
+	pkgi := &kapppkgiv1alpha1.PackageInstall{}
+	objectKey := client.ObjectKey{Namespace: pkgiNamespace, Name: pkgiName}
+
+	// Note: use of the client (remote or not) depends on whether the corresponding cluster is
+	// workload or management which is determined in the Reconcile function
+	if err := c.Get(ctx, objectKey, pkgi); err != nil {
```

Comment:
question: the comment and L239 seem orthogonal.

Reply:
will remove